spouse/spouse_demo.py

```diff
@@ -0,0 +1,552 @@
+#
+# For example, we observe that when mentions of person names occur far apart in a sentence, this is a good indicator that the candidate's label is False. You can write a labeling function that uses preprocessor `get_text_between` or `get_between_tokens`!
+#
+# **IMPORTANT** Good labeling functions manage a trade-off between high coverage and high precision. When constructing your dictionaries, think about building larger, noiser sets of terms instead of relying on 1 or 2 keywords. Sometimes a single word can be very predictive (e.g., `ex-wife`) but it's almost always better to define something more general, such as a regular expression pattern capturing _any_ string with the `ex-` prefix.
+#
+# **Try editing and running the cells below!**
+
+# %%
+# @labeling_function()
+# def lf_new(x):
+#     return POSITIVE if x.person1_word_idx[0] > 3 else ABSTAIN
+#
+# applier = PandasLFApplier([lf_new])
+
+# %%
+# new_dev_L = applier.apply(dev_df)
+# dev_L = np.concatenate((dev_L, new_dev_L), axis=1)
+#
+# new_train_L = applier.apply(train_df)
+# train_L = np.concatenate((train_L, new_train_L), axis=1)
+
+# %% [markdown]
+# ## Part 3: Training the Label Model
+#
+# Now, we'll train a model of the LFs to estimate their accuracies. Once the model is trained, we can combine the outputs of the LFs into a single, noise-aware training label set for our extractor. Intuitively, we'll model the LFs by observing how they overlap and conflict with each other.
+
+# %% [markdown]
+# ## Label Model
+# We know that our labeling functions will not be perfect, and some may be quite low-quality, so we will _model_ their accuracies with a factor-graph based label model, which Snorkel will help us easily apply.
+#
+# This will ultimately produce a single set of **noise-aware training labels**, which are probabilistic or confidence-weighted labels. We will then use these labels to train an end extraction model in the next notebook.  For more technical details of this overall approach, see our [NeurIPS 2016](https://arxiv.org/abs/1605.07723) and [AAAI 2019](https://arxiv.org/abs/1810.02840) papers.
+
+# %% [markdown]
+# ### 1. Training the Model
+# When training the generative model, we use convert_labels with arguments `plusminus` and `categorical` to map our labeling convention from `{0,-1,1}` in the labeling functions to `{0,1,2}` as required by the `LabelModel`.
+#
+# **Parameter Definitions**
+#
+#     cardinality  Cardinality, or the number of classes in the task
+#     lr  The factor by which we update model weights after computing the gradient
+#     class_balance Proportion of [positive, negative] samples in the dataset
+#     n_epochs     A single pass through all the data in your training set
+
+# %%
+from snorkel.labeling.model.label_model import LabelModel
+
+label_model = LabelModel(cardinality=2, verbose=True)
+label_model.fit(train_L, dev_labels, l2=0.0, n_epochs=5000, log_freq=500, seed=123)
```

Comment:
l2=0.0 by default, is this to show the parameter exists?

Reply:
Removed l2.